test/e2e/user_workload_monitoring_test.go

```diff
@@ -819,6 +844,11 @@ func assertGRPCTLSRotation(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+
+	got := countGRPCSecrets(f.Ns) + countGRPCSecrets(f.UserWorkloadMonitoringNs)
```

Comment:
This needs retries because the main `grpc-tls` secret might have been rotated but not yet propagated to the other secrets.